dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -184,14 +183,8 @@ public ReferenceCache getReferenceCache() {
      */
     @Override
     public void initialize() {
-        if (initialized.get()) {
-            return;
-        }
         // Ensure that the initialization is completed when concurrent calls
-        synchronized (startLock) {
-            if (initialized.get()) {
-                return;
-            }
+        if (initialized.compareAndSet(false, true)) {
             // register shutdown hook
```

Comment:
The purpose of the `startLock` is:
> Ensure that the initialization is completed when concurrent calls

If change to `compareAndSet`, it will not block until the `DefaultApplicationDeployer` was initialized, it is unsafe.

Possible case:
1. The first call, start to initialize, but not complete
2. In the second call, check that it has been initialized, skip it, and proceed to the following startup. At this time, the initialization may not be completed, resulting in an error

Reply:
thanks, i got it.
Adjust the type of initialized to volatile boolean